macos/FluentUI/DynamicColor/DynamicColor.swift

```diff
@@ -28,7 +28,7 @@ public class DynamicColor: NSObject {
 	}
 
 	/// resolves color based on theme
-	@objc func resolvedColor(_ appearance: NSAppearance? = nil) -> NSColor {
+	@objc public func resolvedColor(_ appearance: NSAppearance? = nil) -> NSColor {
```

Comment:
can you give me context why this is public now?

Reply:
This logic could be reused in devmain to detect theme changes instead of duplicating code in devmain.

Reply:
I think we have similar code in devmain that already does this. Unless there was a specific scenario that needed this, I would not make it public unless necessary.

Reply:
Yes! Let's not make it public or @objc visible unless it's needed!
This will always have an impact on the binary size.

Reply:
I've reached out to Saad offline to see if there is any alternative. This method is required in devmain to retrieve theme variant of NSColor color.

Reply:
@Saadnajmi  do you have any objections to exposure to these methods? This method is required for because cocoaui consumes color API for their avatar view rendering

Reply:
If we have to, then we have to 🤷🏾
I would file a bug on cocoaui to make sure they request the color on light/dark changes, even if we provide the right colors for them with this API regardless

Reply:
I discussed offline with @Saadnajmi and @rdeassis  offline the possibility of using asset catalog to resolve colors. 
https://developer.apple.com/documentation/xcode/supporting_dark_mode_in_your_interface?language=objc

This would increase binary size as we need to duplicate colors. I am resolving this conversation and I have created another ADO task to continue this discussion. Please feel free to reopen if there are other concerns with the use of `DynamicColor`